chains/evm/cli/admin/withdraw.go

```diff
@@ -14,99 +26,122 @@ var withdrawCmd = &cobra.Command{
 	PreRun: func(cmd *cobra.Command, args []string) {
 		logger.LoggerMetadata(cmd.Name(), cmd.Flags())
 	},
-	Run: withdraw,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		return WithdrawCmd(cmd, args, evmtransaction.NewTransaction, &evmgaspricer.LondonGasPriceDeterminant{})
+	},
+	Args: func(cmd *cobra.Command, args []string) error {
+		err := ValidateWithdrawCmdFlags(cmd, args)
+		if err != nil {
+			return err
+		}
+
+		err = ProcessWithdrawCmdFlags(cmd, args)
+		if err != nil {
+			return err
+		}
+		return nil
+	},
 }
 
-func BindWithdrawFlags(cmd *cobra.Command) {
-	cmd.Flags().StringVar(&Amount, "amount", "", "token amount to withdraw. Should be set or ID or amount if both set error will occur")
-	cmd.Flags().StringVar(&TokenID, "tokenId", "", "token ID to withdraw. Should be set or ID or amount if both set error will occur")
-	cmd.Flags().StringVar(&Bridge, "bridge", "", "bridge contract address")
-	cmd.Flags().StringVar(&Handler, "handler", "", "handler contract address")
-	cmd.Flags().StringVar(&Token, "token", "", "ERC20 or ERC721 token contract address")
-	cmd.Flags().StringVar(&Recipient, "recipient", "", "address to withdraw to")
-	cmd.Flags().Uint64Var(&Decimals, "decimals", 0, "ERC20 token decimals")
-	flags.MarkFlagsAsRequired(cmd, "amount", "tokenId", "bridge", "handler", "token", "recipient", "decimals")
+func BindWithdrawCmdFlags() {
+	withdrawCmd.Flags().StringVar(&Amount, "amount", "", "token amount to withdraw. Should be set or ID or amount if both set error will occur")
+	withdrawCmd.Flags().StringVar(&TokenID, "tokenId", "", "token ID to withdraw. Should be set or ID or amount if both set error will occur")
+	withdrawCmd.Flags().StringVar(&Bridge, "bridge", "", "bridge contract address")
+	withdrawCmd.Flags().StringVar(&Handler, "handler", "", "handler contract address")
+	withdrawCmd.Flags().StringVar(&Token, "token", "", "ERC20 or ERC721 token contract address")
+	withdrawCmd.Flags().StringVar(&Recipient, "recipient", "", "address to withdraw to")
+	withdrawCmd.Flags().Uint64Var(&Decimals, "decimals", 0, "ERC20 token decimals")
+	flags.MarkFlagsAsRequired(withdrawCmd, "amount", "tokenId", "bridge", "handler", "token", "recipient", "decimals")
 }
 
 func init() {
-	BindWithdrawFlags(withdrawCmd)
+	BindWithdrawCmdFlags()
```

Comment:
I think we should stick with the pattern of `BindWithdrawCmFlags` getting a command as an argument - we use this throughout codebase (and it allows reusage of these bind functions)